### PR TITLE
950 - TYDE CoS #4 - Create communities tab

### LINKED ISSUE #950: CoS #4 - Create communities tab

### PR DESCRIPTION
**Pull request by narendrasss**
## Changes
* Implement the add communities tab of the TYDE Circle of Support activity

## Screenshots
https://user-images.githubusercontent.com/31267630/109707174-b50b1500-7b4e-11eb-9491-78c10a185a4a.mp4

https://user-images.githubusercontent.com/31267630/107430271-c4f88180-6ad9-11eb-8e38-77ab2115201b.mp4

## Issue Linkage
Closes #950 

## PR Dependency
Depends on: #948, #960

## Automated Testing
N/A

### CODE REVIEW

**Comment by narendrasss** [comment]
@wynnset This is still a WIP, but can you see if the tab looks good? Want your approval since it's a bit different from the figma design.

**Comment by wynnset** [comment]
Could you please make the "Add Community" button a bit more prominent? Also, we should disable closing the panel using the down icon you use at the end of your video. It should be disabled as soon as they change anything in the community panel.

**Comment by narendrasss** [comment]
> Also, we should disable closing the panel using the down icon you use at the end of your video.

Hmm I'm curious why we would want to do this. I think intuitively it makes sense to cancel by pressing the same button to slide up. Currently, any changes the user makes when they close the tab _is kept_, meaning when they open the tab again it'll be where they left off. I can change that, but in that case maybe an alert would be a better approach?

**Comment by narendrasss** [comment]
> Could you please make the "Add Community" button a bit more prominent?

Increased the font size a bit:

CleanShot 2021-02-04 at 11 58 11@2x

Does that look ok?

**Comment by wynnset** [comment]
> @wynnset Hmm I'm curious why we would want to do this. I think intuitively it makes sense to cancel by pressing the same button to slide up. Currently, any changes the user makes when they close the tab _is kept_, meaning when they open the tab again it'll be where they left off. I can change that, but in that case maybe an alert would be a better approach?

Yeah, sure that makes sense. I think the alert approach would be good. I'm just afraid that they might assume that closing the panel will automatically add it. We can confirm with them if they want to add it e.g. "Would you like to add this community now or later?" or something along those lines.

**Comment by wynnset** [comment]
> > Could you please make the "Add Community" button a bit more prominent?
> 
> Increased the font size a bit:
> 
> CleanShot 2021-02-04 at 11 58 11@2x
> 
> Does that look ok?

Can we have it so it's not outlined and gray? Keep the color consistent with other Vue elements e.g. NodeModal submit button.

**Comment by narendrasss** [comment]
> Can we have it so it's not outlined and gray? Keep the color consistent with other Vue elements e.g. NodeModal submit button.

I wanted to keep it consistent with the connections tab:

CleanShot 2021-02-09 at 10 41 18@2x

Or should I change that as well? I wanted the button to be consistent with the CoS design, not necessarily the NodeModal design.

**Comment by wynnset** [comment]
> I wanted to keep it consistent with the connections tab
> Or should I change that as well? I wanted the button to be consistent with the CoS design, not necessarily the NodeModal design.

I think the connections tab is OK. We could change that if you want to - I don't have a preference there. I just think the main submit button needs to be more prominent as the users may not realize it's a button or notice it. Usually, the submit button should be the most prominent thing on the UI.

**Comment by narendrasss** [comment]
Made the changes:

https://user-images.githubusercontent.com/31267630/107430094-98446a00-6ad9-11eb-85ed-cb146707a099.mp4

@wynnset Let me know if there's anything else you want to change re: this ticket.

**Comment by wynnset** [comment]
Some comments from testing needing to be addressed:
- "Would you like to add this community now or later?" should only pop up if the user has entered at least the name of the connection
- Should allow adding a connection to more than 1 community (currently limited to 1)
- When I refresh the page, any connections or communities I added before the refresh disappears
- Similar to the connections issue, the emoji selector doesn't allow me to select an emoji (just shows the list of emojis)

**Review by shirleywbi — COMMENTED** [review]
@narendrasss 
image
1. Should user be able to CTRL+A to select text at (1)
2. Padding?
3. The grey buttons make me think that they're disabled. Should we change the color? Also, my view looks different from the video. Not sure if that's intended or just a Windows Chrome view.

**Comment by narendrasss** [comment]
@shirleywbi thanks, padding should be fixed!

I think the issue with (1) is present with text inputs in general because of the "select all nodes" function. Maybe have it as a separate ticket?

File: templates/vue/src/components/tyde/activities/CircleOfSupport/AddCommunityForm.vue
Comment (shirleywbi) [inline templates/vue/src/components/tyde/activities/CircleOfSupport/AddCommunityForm.vue:62]: Should a default color be set? Otherwise, color is stored as "" in the db.

**Review by shirleywbi — COMMENTED** [review]
@narendrasss Communities tab is working fine. Do we need to be able to update/remove communities?

Also, is this just me? I wasn't able to test any of the actions corresponding to connections since nothing is showing up for me for Connections: image

**Comment by narendrasss** [comment]
> Also, is this just me? I wasn't able to test any of the actions corresponding to connections since nothing is showing up for me for Connections:

Should be fixed now.

File: templates/vue/src/components/tyde/activities/CircleOfSupport/AddCommunityForm.vue
Comment (narendrasss) [reply templates/vue/src/components/tyde/activities/CircleOfSupport/AddCommunityForm.vue:62]: @wynnset thoughts?

**Review by shirleywbi — APPROVED** [review]
LGTM!

File: templates/vue/src/components/tyde/activities/CircleOfSupport/ConnectionsTab/AddConnectionForm.vue
Comment (wynnset) [inline templates/vue/src/components/tyde/activities/CircleOfSupport/ConnectionsTab/AddConnectionForm.vue:37]: This should read "Which communities does this person belong to?"

Comment (narendrasss) [reply templates/vue/src/components/tyde/activities/CircleOfSupport/ConnectionsTab/AddConnectionForm.vue:37]:
```suggestion
        <h1 class="community-title">Which communities does this person belong to?</h1>
```

**Comment by wynnset** [comment]
I've reviewed most of the code here, but it's hard to complete the review without #960 changes since some of this work here is half-working. I'll wait for #960 to get merged in before merging this one in so I can review everything at the same time.

**Review by wynnset — APPROVED** [review]

**Comment by wynnset** [comment]
This is ready for merge, but want to wait for the merge conflicts in #916 to be resolved before merging this in

**Pull request merged**
